calyx-ir/src/rewriter.rs

```diff
@@ -10,7 +12,7 @@ pub type RewriteMap<T> = HashMap<ir::Id, RRC<T>>;
 
 /// Map to rewrite port uses. Maps the canonical name of an old port (generated using
 /// [ir::Port::canonical]) to the new [ir::Port] instance.
-pub type PortRewriteMap = HashMap<ir::Canonical, RRC<ir::Port>>;
+pub type PortRewriteMap = LinkedHashMap<ir::Canonical, RRC<ir::Port>>;
```

Comment:
I suppose this is the straightforward way to get determinism for all this rewriting… we have, of course, run into this nondeterminism before, so "someday" it might be nice to revisit the contract that the rewriter has with its clients.